include/lbann/models/model.hpp

```diff
@@ -49,71 +49,54 @@ namespace lbann {
 // Forward declarations
 class lbann_callback;
 
-/** Base class for neural network models. */
+/** @brief Abstract base class for neural network models. */
 class model {
 public:
 
-  model(lbann_comm *comm,
+  // ===========================================
+  // Life cycle functions
+  // ===========================================
+
+  model(lbann_comm* comm,
         El::Int mini_batch_size,
-        objective_function *obj_fn,
+        objective_function* obj_fn,
```

Comment:
If we're gonna move the `*` around here, we should be consistent (at least in this file).